Mathlib/Data/Nat/Dist.lean

```diff
@@ -78,8 +78,8 @@ theorem dist_zero_left (n : ℕ) : dist 0 n = n :=
 theorem dist_add_add_right (n k m : ℕ) : dist (n + k) (m + k) = dist n m :=
   calc
     dist (n + k) (m + k) = n + k - (m + k) + (m + k - (n + k)) := rfl
-    _ = n - m + (m + k - (n + k)) := by rw [add_tsub_add_eq_tsub_right]
-    _ = n - m + (m - n) := by rw [add_tsub_add_eq_tsub_right]
+    _ = n - m + (m + k - (n + k)) := by rw [@add_tsub_add_eq_tsub_right]
+    _ = n - m + (m - n) := by rw [@add_tsub_add_eq_tsub_right]
```

Comment:
This looks very weird; what goes wrong without the `@`?

Reply:
It can't find some instance. This is one of the main two issues in this PR.

Reply:
I think this would be worth opening a Lean issue about, along with a TODO comment referencing it; that way, we can more revert this change later if Lean fixes it.

Reply:
Yes, here is an explanation what is going wrong: https://github.com/leanprover/lean4/pull/2003#issuecomment-1382611497